flink-table-store-connector/src/test/java/org/apache/flink/table/store/connector/FileStoreITCase.java

```diff
@@ -0,0 +1,197 @@
+import static org.apache.flink.table.store.file.FileStoreOptions.FILE_FORMAT;
+import static org.apache.flink.table.store.file.FileStoreOptions.FILE_PATH;
+import static org.assertj.core.api.Assertions.assertThat;
+
+/** ITCase for {@link FileStoreSource} and {@link StoreSink}. */
+@RunWith(Parameterized.class)
+public class FileStoreITCase extends AbstractTestBase {
+
+    private static final RowType PARTITION_TYPE =
+            new RowType(Collections.singletonList(new RowType.RowField("p", new VarCharType())));
+
+    private static final RowType KEY_TYPE =
+            new RowType(Collections.singletonList(new RowType.RowField("k", new IntType())));
+
+    private static final RowType VALUE_TYPE =
+            new RowType(
+                    Arrays.asList(
+                            new RowType.RowField("v", new IntType()),
+                            new RowType.RowField("p", new VarCharType()),
+                            // rename key
+                            new RowType.RowField("_k", new IntType())));
+
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    private static final DataStructureConverter<RowData, Row> CONVERTER =
+            (DataStructureConverter)
+                    DataStructureConverters.getConverter(
+                            TypeConversions.fromLogicalToDataType(VALUE_TYPE));
+
+    private static final int NUM_BUCKET = 3;
+
+    private static final List<RowData> SOURCE_DATA =
+            Arrays.asList(
+                    wrap(GenericRowData.of(0, StringData.fromString("p1"), 1)),
+                    wrap(GenericRowData.of(0, StringData.fromString("p1"), 2)),
+                    wrap(GenericRowData.of(5, StringData.fromString("p1"), 1)),
+                    wrap(GenericRowData.of(6, StringData.fromString("p2"), 1)),
+                    wrap(GenericRowData.of(3, StringData.fromString("p2"), 5)),
+                    wrap(GenericRowData.of(5, StringData.fromString("p2"), 1)));
+
+    private Configuration options;
+
+    private final boolean isBounded;
+
+    public FileStoreITCase(boolean isBounded) {
+        this.isBounded = isBounded;
+    }
+
+    @Parameterized.Parameters(name = "isBounded-{0}")
+    public static List<Boolean> getVarSeg() {
+        return Arrays.asList(true, false);
+    }
+
+    private static SerializableRowData wrap(RowData row) {
+        return new SerializableRowData(row, InternalSerializers.create(VALUE_TYPE));
+    }
+
+    @Before
+    public void before() throws IOException {
+        options = new Configuration();
+        options.set(BUCKET, NUM_BUCKET);
+        options.set(FILE_PATH, TEMPORARY_FOLDER.newFolder().toURI().toString());
+        options.set(FILE_FORMAT, "avro");
+    }
+
+    @Test
+    public void testPartitioned() throws Exception {
+        innerTest(true);
+    }
+
+    @Test
+    public void testNonPartitioned() throws Exception {
+        innerTest(false);
+    }
+
+    private void innerTest(boolean partitioned) throws Exception {
```

Comment:
What about failure tests? We need checkpoints and failures to make sure this 2pc stuff actually works and guarantees exactly-once semantics.

Also what about overwrite tests?

Reply:
How to write failure tests in ITCase? These should exist in `GlobalCommitterOperatorTest`?

Reply:
I can add some overwrite tests

Reply:
> How to write failure tests in ITCase? These should exist in `GlobalCommitterOperatorTest`?

For example use `FailingAtomicRenameFileSystem` to trigger failures with probability. Allow the job to restart from failure and check if the final result follows the exactly-once semantics. This is one of the scenario we want to address so it would be better to have it in IT case.

Reply:
Add a test for asserting no impacting when failover

Reply:
> use FailingAtomicRenameFileSystem to trigger failures with probability

If there is a problem with FileSystem, it can only happen when the file starts to be written with an exception and nothing happens yet. What's the point of this?
I don't think ITCase is response to test failover.

Reply:
You counld find a good example in Flink code.